tests/test_group.py

```diff
@@ -6,6 +6,7 @@
 
 import pytest
 import time
+import uuid
```

Comment:
We should move this to the first block of imports so it complies with PEP8. Also, we can probably remove `import time` (which is also in the wrong place) once the other uses are replaced.